pdo_sqlsrv/core_results.cpp

```diff
@@ -335,13 +338,30 @@ sqlsrv_buffered_result_set::sqlsrv_buffered_result_set( sqlsrv_stmt* stmt TSRMLS
                 meta[i].length += sizeof( char ) + sizeof( SQLULEN ); // null terminator space
                 offset += meta[i].length;
                 break;
+            case SQL_CHAR:
```

Comment:
If encoding is set to utf8, we would be calling SQLGetData with target c type set to SQL_C_WCHAR.